### PR TITLE
CLI: Add version to Jetpack Release

### PR DESCRIPTION
**Pull request by sdixon194**
#### Changes proposed in this Pull Request:

Adds `jetpack release [project] version` to the CLI, which lets you bump the version in composer.json, package.json and the main plugin's php file. It also auto suggests versions for you.

#### Jetpack product discussion

N/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

Try using `jetpack release` and following the prompts. 
Try passthroughs, such as `jetpack release plugins/jetpack version`
Try using some flags:
-a (alpha)
-b (beta)
-s (stable)
e.g. `jetpack release plugins/jetpack version -a` should prompt the next alpha release.

`plugins/jetpack` alpha releases should be in format x.y-a.z, all other should be x.y-alpha.

### CODE REVIEW

File: tools/cli/commands/release.js [hunk 1]
```diff
@@ -230,33 +247,80 @@ export async function getReleaseVersion( argv ) {
...
+		} else {
+			stableVersion = await getVersionBump( stableVersion );
+			potentialVersion = `${ stableVersion }-alpha`;
+		}
+		return potentialVersion;
+	}
 }
 
+/**
+ * Bumps the correct number.
```
Comment (anomiex) [inline tools/cli/commands/release.js:283]: This does a bump for when composer.json has `.extra.changelogger.versioning` set to 'wordpress'. It gets it wrong for the default 'semver'.

File: tools/cli/commands/release.js [hunk 2]
```diff
@@ -269,17 +333,29 @@ export async function promptForVersion( argv, version ) {
...
-	argv.beta = response.beta;
-	argv.b = response.beta;
+
+	switch ( response.version_type ) {
+		case 'beta':
+			argv.b = true;
+			argv.beta = true;
+			break;
+		case 'alpha (including Atomic)':
+			argv[ 'dev-version' ] = true;
```
Comment (anomiex) [inline tools/cli/commands/release.js:352]: As above
```suggestion
			argv.devRelease = true;
```

File: tools/cli/commands/release.js (hunk 1 above)
Comment (sdixon194) [reply tools/cli/commands/release.js:284]: Alright, tried to clean this one up as best I could!

File: tools/cli/commands/release.js [hunk 3]
```diff
@@ -260,19 +260,20 @@ export async function getReleaseVersion( argv ) {
...
-	if ( argv.a || argv[ 'dev-version' ] ) {
-		// Jetpack uses additional versioning for dev/atomic in the form of x.y-a.z
-		if ( argv.project === 'plugins/jetpack' ) {
-			if ( alphaVersion ) {
-				alphaVersion = await getVersionBump( alphaVersion );
-				potentialVersion = `${ stableVersion }-${ alphaVersion }`;
+	// Handle alpha/dev-release version if necessary.
+	if ( argv.a || argv.devRelease ) {
+		// Check if dev-releases is specified in project's composer.json
+		const hasDevReleases = await readComposerJson( argv.project ).extra[ 'dev-releases' ];
```
Comment (anomiex) [inline tools/cli/commands/release.js:266]: I was going to suggest using optional chaining to handle the case where `extra` isn't set in composer.json, like
```js
		const hasDevReleases = await readComposerJson( argv.project ).extra?.[ 'dev-releases' ];
```
except the `esm` package we're using breaks on that syntax and [isn't being updated](https://href.li/?https://github.com/standard-things/esm/issues/866). Sigh.

File: tools/cli/commands/release.js (hunk 1 above)
Comment (anomiex) [reply tools/cli/commands/release.js:284]: Looks good to me!

File: tools/cli/commands/release.js [hunk 4]
```diff
@@ -283,31 +284,43 @@ export async function getReleaseVersion( argv ) {
...
 	}
 
-	if ( version[ 1 ] === '9' ) {
-		// e.g, if the current version is 10.9 and we want 11.0
-		version[ 0 ] = parseInt( version[ 0 ] ) + 1;
-		version[ 1 ] = '0';
-	} else {
-		// e.g, if the current version is 10.8 and we want 10.9
-		version[ 1 ] = parseInt( version[ 1 ] ) + 1;
+	const versioningType = await readComposerJson( project ).extra.changelogger;
```
Comment (anomiex) [inline tools/cli/commands/release.js:299]: `versioningType` isn't really a correct name. `changeloggerConfig` maybe?

Even better would be to use optional chaining, as above, but again `esm` breaks.
```js
	const versioningType = await readComposerJson( project ).extra?.changelogger?.versioning;
```

**Review by anomiex — APPROVED** [review]
One variable naming nitpick inline, otherwise looks good to me.

File: tools/cli/commands/release.js (hunk 3 above)
Comment (sdixon194) [reply tools/cli/commands/release.js:266]: Ah that woulda been slick.

File: tools/cli/commands/release.js (hunk 4 above)
Comment (sdixon194) [reply tools/cli/commands/release.js]: Good idea, updated the variable name 👍

**Review by anomiex — APPROVED** [review]

**Pull request merged**